LoopKit/Pluggable.swift

```diff
@@ -0,0 +1,22 @@
+//
+//  Pluggable.swift
+//  LoopKit
+//
+//  Created by Nathaniel Hamming on 2023-09-08.
+//  Copyright © 2023 LoopKit Authors. All rights reserved.
+//
+
+public protocol Pluggable: AnyObject {
+    /// The unique identifier for this plugin.
+    static var pluginIdentifier: String { get }
+    
+    /// A plugin may need a reference to another plugin. This callback allows for such a reference.
+    /// It is called once during app initialization after plugins are initialized and again as new plugins are added and initialized.
+    func initializationComplete(for plugins: [Pluggable])
```

Comment:
nit: I like the `Pluggable` term because it's differentiated from "Plugin"  A loop "Plugin" is a framework/module that vends types that conform to `Pluggable`, and Loop can have (in theory) multiple `Pluggable` instances of types the plugin defines.  Just wondering if may this method signature should be `func initializationComplete(for pluggables: [Pluggable])`.

Reply:
will add in a clean up commit